frame/bridge/relayer-game/src/lib.rs

```diff
@@ -264,581 +364,352 @@ impl<T: Trait<I>, I: Instance> Module<T, I> {
-			{
-				if maybe_confirmed_proposal.is_none() {
-					maybe_confirmed_proposal = Some(proposal.to_owned());
-				} else {
-					error!("Honest Relayer Count - MORE THAN 1 WITHIN A ROUND");
-				}
-			} else {
-				evils.push((
-					proposal.relayer.clone(),
-					proposal.bonded_proposal.last().unwrap().0,
-				));
-			}
-		}
+		// Check if the proposed header has already been relaied
+		ensure!(
+			game_id > last_confirmed_block_id,
+			<Error<T, I>>::RelayStuffsAC
+		);
+		// Make sure the game is at first round
+		ensure!(
+			Self::proposals_of_game_at(&game_id, 1).is_empty(),
+			<Error<T, I>>::RoundMis
+		);
 
-		if let Some(confirmed_proposal) = maybe_confirmed_proposal {
-			let rewards = evils
-				.into_iter()
-				.map(|evil| {
-					(
-						(
-							confirmed_proposal.relayer.clone(),
-							confirmed_proposal.bonded_proposal.last().unwrap().0,
-						),
-						evil,
-					)
-				})
-				.collect();
-
-			Self::settle_with_challenge(last_round, proposals, &confirmed_proposal, rewards);
-
-			// TODO: reward if no challenge
-
-			Some(confirmed_proposal.bonded_proposal[0].1.hash())
-		} else {
-			info!(target: "relayer-game", "   >  No Honest Relayer");
+		let existed_proposals = Self::proposals_of_game_at(&game_id, 0);
+		let proposal_content = vec![parcel];
 
-			Self::settle_abandon(last_round_proposals);
+		if existed_proposals.is_empty() {
+			// A new game might open
+
+			// Check if it is ok to open more games
+			ensure!(
+				active_games < T::RelayerGameAdjustor::max_active_games(),
+				<Error<T, I>>::ActiveGamesTM
+			);
+		} else {
+			// An against proposal might add
 
-			None
+			ensure!(
+				Self::is_game_open_at(&game_id, <frame_system::Module<T>>::block_number()),
+				<Error<T, I>>::GameC
+			);
+			ensure!(
+				Self::is_unique_proposal(&proposal_content, &existed_proposals),
+				<Error<T, I>>::ProposalDup
+			);
 		}
-	}
 
-	pub fn update_samples(game_id: GameId<TcBlockNumber<T, I>>) {
-		<Samples<T, I>>::mutate(game_id, |samples| {
-			T::RelayerGameAdjustor::update_samples(samples);
+		let bond = Self::ensure_can_bond(&relayer, 0, existed_proposals.len() as u8 + 1)?;
+		// let proposal = Self::build_proposal(relayer, samples, proofses, bond)?;
+		let proposal = {
+			let mut proposal = RelayProposal::new();
 
-			if samples.len() < 2 {
-				error!("Sample Points MISSING, Check Your Sample Strategy Implementation");
+			proposal.relayer = relayer;
+			proposal.parcels = proposal_content;
+			proposal.bond = bond;
 
-				return;
+			// Allow propose without proofs
+			// The proofs can be completed later through `complete_proofs`
+			if let Some(proofs) = proofs {
+				T::RelayableChain::verify_proofs(&proposal.parcels[0], &proofs)?;
+
+				proposal.verified = true;
 			}
 
-			Self::deposit_event(RawEvent::NewRound(
-				game_id,
-				samples.concat(),
-				samples[samples.len() - 1].clone(),
-			));
-		});
-	}
+			proposal
+		};
+		let round = 0;
+
+		<ActiveGames<I>>::mutate(|count| *count += 1);
```

Comment:
Why auto increase the counter if it is joining an existing game?

Reply:
Can't get the active games count without this, unless iterate the propsals but that's too expensive

Reply:
And `propose` only accept new block

If join an existing game, must call `extend_proposal`